web/js/duplicates.js

```diff
@@ -188,6 +194,7 @@ $(function() {
     // Want to show potential duplicates when a regular user starts a new
     // report, or changes the category/location of a partial report.
     $("#problem_form").on("change.category", "select#form_category", problem_form_category_change);
+    $("select#form_category").change(); // In case on page load something is selected
```

Comment:
I wonder whether this line should be in a separate commit to the other changes?

This line does the actual " Look up duplicate in case onload."

Whereas the other lines—I think?—fix a different bug, which was that the `nearby_url` API was getting queried even if `$('select[name="category"]').val()` was `-- Pick a category --` (which was obviously inefficient, so makes sense to resolve the defer immediately, as you’ve now done).

Or have I missed something?

Reply:
Yeah, could be; I only noticed because of it happening onload after this change, but it could happen if someone changed back to Pick as well, yes.